pinot-common/src/test/java/org/apache/pinot/common/utils/FileUtilsTest.java

```diff
@@ -0,0 +1,125 @@
+package org.apache.pinot.common.utils;
+
+import com.google.common.collect.Iterables;
+import junit.framework.TestCase;
+import org.junit.jupiter.api.Test;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FileUtilsTest extends TestCase {
+    @Test
+    public void testMoveFileWithOverwrite_destFileNotPresent() {
+        // Create source directory
+        File sourceDir = new File("/tmp/sourceDir");
+        sourceDir.mkdir();
+        // Create destination directory
+        File destDir = new File("/tmp/destDir");
+        destDir.mkdir();
+
+        // Define source and dest file
+        File sourceFile = new File(sourceDir, "sourceFile.txt");
+        try {
+            // Create empty source file
+            sourceFile.createNewFile();
+            FileUtils.moveFileWithOverwrite(sourceFile, new File(destDir, "destFile.txt"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        String[] fileList = destDir.list();
+        for(String file : fileList) {
```

Comment:
Is a linear search warranted? Can we not just try to open the file in the destDir, and assert we can open it? It would be good to also ensure something like file size is the same (if not content).